dnf-repoconfig-daemon/repoconf.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "configuration.hpp"
+#include "repoconf.hpp"
+
+#include <iostream>
+#include <sdbus-c++/sdbus-c++.h>
+#include <string>
+
+
+RepoConf::RepoConf(sdbus::IConnection &connection) : connection(connection)
+{
+    dbus_register_methods();
+}
+
+void RepoConf::dbus_register_methods()
+{
+    const std::string objectPath = "/org/rpm/dnf/v0/rpm/RepoConf";
+    const std::string interfaceName = "org.rpm.dnf.v0.rpm.RepoConf";
+
+    dbus_object = sdbus::createObject(connection, objectPath);
+    dbus_object->registerMethod(interfaceName, "list", "as", "aa{sv}", [this](sdbus::MethodCall call) -> void {this->list(call);});
+    dbus_object->registerMethod(interfaceName, "get", "s", "a{sv}", [this](sdbus::MethodCall call) -> void {this->get(call);});
+    dbus_object->registerMethod(interfaceName, "enable", "as", "as", [this](sdbus::MethodCall call) -> void {this->enable_disable(call, true);});
+    dbus_object->registerMethod(interfaceName, "disable", "as", "as", [this](sdbus::MethodCall call) -> void {this->enable_disable(call, false);});
```

Comment:
Matter of taste, but maybe merge this into `SetEnabled(a(sb))`? Also, is it really much of a use case to enable or disable several repos at once? I. e. maybe a `SetEnabled(sb)` is just enough? This is interesting in the failure case -- would some repositories still be enabled/disabled even if another one fails?

Why do these return a string array? If the enablement fails, they should fail with a DBus error, and if it succeeds, the state is predictable. Also, the `void` smells like it doesn't *actually* return anything?